server/src/main/java/org/apache/druid/metadata/segment/cache/SegmentRecord.java

```diff
@@ -96,8 +97,8 @@ static SegmentRecord fromResultSet(ResultSet r)
   }
 
   @Nullable
-  private static DateTime nullSafeDate(String date)
+  private static DateTime nullAndEmptySafeDate(String date)
```

Comment:
Since a similar logic is being used elsewhere too, I wonder if we shouldn't just make this a static utility method in `SqlSegmentsMetadataQuery`.

Reply:
Oops I merged and missed this comment.